apps/wallet-mobile/src/features/ReviewTx/common/operations.tsx

```diff
@@ -0,0 +1,201 @@
+
+  return (
+    <View style={styles.operation}>
+      <Text style={styles.operationLabel}>{strings.rewardsWithdrawalLabel}</Text>
+
+      <Text style={styles.operationValue}>{strings.rewardsWithdrawalText}</Text>
+    </View>
+  )
+}
+
+export const StakeDelegateOperation = ({poolId}: {poolId: string}) => {
+  const {styles} = useStyles()
+  const strings = useStrings()
+  const poolInfo = usePoolInfo({poolId})
+  const {networkManager} = useSelectedNetwork()
+
+  const poolInfoText = poolInfo != null ? `[${poolInfo.ticker}] ${poolInfo.name}` : poolId
+
+  return (
+    <View style={styles.operation}>
+      <Text style={styles.operationLabel}>{strings.delegateStake}</Text>
+
+      <Space width="lg" />
+
+      <TouchableOpacity
+        activeOpacity={0.5}
+        onPress={() => Linking.openURL(networkManager.explorers.cardanoscan.pool(poolId))}
+      >
+        <Text style={styles.operationLink}>{poolInfoText}</Text>
+      </TouchableOpacity>
+    </View>
+  )
+}
+
+export const AbstainOperation = () => {
+  const {styles} = useStyles()
+  const strings = useStrings()
+
+  return (
+    <View style={styles.operation}>
+      <Text style={styles.operationLabel}>{strings.selectAbstain}</Text>
+    </View>
+  )
+}
+
+export const NoConfidenceOperation = () => {
+  const {styles} = useStyles()
+  const strings = useStrings()
+
+  return (
+    <View style={styles.operation}>
+      <Text style={styles.operationLabel}>{strings.selectNoConfidence}</Text>
+    </View>
+  )
+}
+
+export const VoteDelegationOperation = ({drepID}: {drepID: string}) => {
+  const {styles} = useStyles()
+  const strings = useStrings()
+
+  const bech32DrepId = useDrepBech32Id(drepID)
+
+  return (
+    <View style={styles.operation}>
+      <Text style={styles.operationLabel}>{strings.delegateVotingToDRep}</Text>
+
+      <Space width="lg" />
+
+      <Text style={styles.operationValue}>{bech32DrepId ?? drepID}</Text>
+    </View>
+  )
+}
+
+export const useOperations = (certificates: FormattedCertificates | null) => {
+  if (certificates === null) return []
+
+  return certificates.reduce<React.ReactNode[]>((acc, [certificateKind, CertificateData], index) => {
+    switch (certificateKind) {
+      case CertificateTypes.StakeRegistration:
+        return [...acc, <StakeRegistrationOperation key={index} />]
+
+      case CertificateTypes.StakeDeregistration:
+        return [...acc, <StakeDeregistrationOperation key={index} />]
+
+      case CertificateTypes.StakeDelegation: {
+        const poolKeyHash = (CertificateData as Certificate[CertificateTypes.StakeDelegation]).pool_keyhash ?? null
+        if (poolKeyHash == null) return acc
+        return [...acc, <StakeDelegateOperation key={index} poolId={poolKeyHash} />]
+      }
+
+      case CertificateTypes.VoteDelegation: {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const drep: any = (CertificateData as Certificate[CertificateTypes.VoteDelegation]).drep
```

Comment:
typescripp gets crazy when you mix two shapes.

drep is `'AlwaysAbstain' | 'AlwaysNoConfidence' |  {KeyHash: string} |  {ScriptHash: string}` ans it comes from CSL

Reply:
But why is it an issue it has that shape? You can just check for it and it doesn't require the `any`

```
        const drep = (CertificateData as Certificate[CertificateTypes.VoteDelegation]).drep

        if (drep === 'AlwaysAbstain') return [...acc, <AbstainOperation key={index} />]
        if (drep === 'AlwaysNoConfidence') return [...acc, <NoConfidenceOperation key={index} />]

        const drepId = ('KeyHash' in drep ? drep.KeyHash : drep.ScriptHash) ?? ''
```

Reply:
yeah ! forgot about this